File: backend-dotnet/Domain/Entities/UserFollow.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Domain.Entities
{
    public class UserFollow
    {
        [Key]
        public Guid Id { get; set; } = Guid.NewGuid();

        [Required]
        public Guid FollowerId { get; set; }

        [ForeignKey("FollowerId")]
        public User Follower { get; set; } = null!;

        [Required]
        public Guid FollowingId { get; set; }

        [ForeignKey("FollowingId")]
        public User Following { get; set; } = null!;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: backend-java/src/main/java/com/jerrygram/domain/entities/UserFollow.java
package com.jerrygram.domain.entities;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "\"UserFollows\"", uniqueConstraints = {
    @UniqueConstraint(columnNames = {"\"FollowerId\"", "\"FollowingId\""})
})
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(exclude = {"follower", "following"})
@ToString(exclude = {"follower", "following"})
public class UserFollow {
    
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "\"Id\"")
    private UUID id;
    
    @Column(name = "\"FollowerId\"", nullable = false)
    private UUID followerId;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "\"FollowerId\"", nullable = false, insertable = false, updatable = false)
    private User follower;
    
    @Column(name = "\"FollowingId\"", nullable = false)
    private UUID followingId;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "\"FollowingId\"", nullable = false, insertable = false, updatable = false)
    private User following;
    
    @CreationTimestamp
    @Column(name = "\"CreatedAt\"", nullable = false)
    private LocalDateTime createdAt;
}